frontend/catalyst/jax_tracer.py

```diff
@@ -417,6 +419,35 @@ def trace_quantum_tape(
     #       equations in a wrong order. The set of variables are always complete though, so we sort
     #       the equations to restore their correct order.
 
+    def _bind_native_controlled_op(qrp, op, controlled_wires, controlled_values):
+        # For named-controlled operations (e.g. CNOT, CY, CZ) - bind directly by name. For
+        # `Controlled(OP)` bind OP with native quantum control syntax.
+        if op.__class__ in {Controlled, ControlledOp, ControlledQubitUnitary}:
+            return _bind_native_controlled_op(qrp, op.base, op.control_wires, op.control_values)
```

Comment:
Does this mean we should remove these lines or no?
https://github.com/PennyLaneAI/catalyst/blob/d191ede947b591a52c399f6f851cd1b2a64a645c/frontend/catalyst/qjit_device.py#L192-L193
I didn't see them in this PR.

Reply:
I forgot about this place and this is this is interesting! PL seems to skip the decomposition if I add the right `C(Op)` into the custom device's spec. This seems to much our needs. See e.g. [here](https://github.com/PennyLaneAI/catalyst/blob/3602c696fc2843e787901b52797e3eeb1e7e7672/frontend/test/lit/test_quantum_control.py#L120).

Reply:
`C(QubitUnitary)` is added to the op support list for lightning devices right?

But I guess this means this patching is still useful to keep as a backup when native support is not available right?